cmd/ztest/ztest.c

```diff
@@ -533,10 +533,16 @@ _umem_logging_init(void)
 static void
 dump_debug_buffer(void)
 {
-	if (!ztest_dump_debug_buffer)
+	ssize_t ret __attribute__((unused));
```

Comment:
I think you can discard the return value with `(void)` as we do elsewhere

Reply:
I tried that. GCC still complains. The only way to get it to shut up wass to save the return value. 🤷‍♂️

Reply:
What's the error from gcc?  `(void)` works in other places in this file (at least, for other functions)

Reply:
```
warning: ignoring return value of ‘ret’, declared with attribute warn_unused_result [-Wunused-result]
```

This a discussion which goes, way back https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425.  The short version is the behavior here isn't clearly defined by the standard and is compiler specific.

Reply:
Huh.  Any idea why all the other `(void)`'s in this file work fine, but it won't work here?

Reply:
It has something to do with the `write()` call and its attributes specifically, I believe.

Reply:
For several years now most of the standard glibc functions have had their prototypes updated to include the [warn_unused_result](https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#Common-Function-Attributes) function attribute, which forces the warning.  Which is why we only see this in some places.

```c
#define __wur __attribute_warn_unused_result__

extern ssize_t write (int __fd, const void *__buf, size_t __n) __wur;
```